core/src/main/scala/tofu/Context.scala

```diff
@@ -80,66 +162,149 @@ object WithContext {
+    *
+    * @example
+    * Example of usage is to hide sensitive information {{{
+    *
+    * import tofu.syntax.context._
+    *
+    * case class UserContext(
+    *              id: String,
+    *              phoneNumber: String //very sensitive
+    *            )
+    * def hidePartOfPhoneNumber: String => String = ???
+    *
+    * def contextualLogInfo[F[_]](message: String)
+    *     (implicit hasUserContext: UserContext In F): F[Unit] =
+    *     ??? //logs both message AND UserContext
+    *
+    * def program[F[_]: Monad: ] = for {
+    *    user <- obtainUserSomehow[F]
+    *    implicit0(hasUserContext: User In F) <- Context.const[F, User](user.toContext)
+    *    logUser =  contextualLogInfo[F](s"Successfully obtained user")
+    *    _ <- logUser.local(hidePartOfPhoneNumber) //logs only part of phone number
+    * } yield ()
+    * }}}
+    * @param fa      computation that is going to run with different context
+    * @param project pure function
+    * @return result of ran computation `fa`
+    */
   def local[A](fa: F[A])(project: Ctx => Ctx): F[A]
 
+  /** Allows to focus [[Local]] on inner parts of its context with lens.
+    *
+    * @param contains lens that can extract from `Ctx` or set some value of type `A`
+    */
   def subcontext[A](contains: Ctx Contains A): WithLocal[F, A] = new LocalContainsInstance[F, Ctx, A](this, contains)
 }
 
 object Local {
   def apply[F[_]](implicit ctx: Local[F]): HasLocal[F, ctx.Ctx] = ctx
+
   type Aux[F[_], C] = HasLocal[F, C]
 }
 
-/** lightweight version of ApplicativeLocal
-  * allows to modify value of `C` locally for some subprocess
-  */
+/** Synonym for [[Local]] with C as Ctx for better type inference */
 trait WithLocal[F[_], C] extends Local[F] with WithContext[F, C]
 
 object WithLocal {
   def apply[F[_], C](implicit ctx: WithLocal[F, C]): WithLocal[F, C] = ctx
 }
 
+/** Allows to evaluate contextual computation with some context
+  *
+  * The main use case for it is to obtain some context `Lower[Ctx]`,
+  * and then to run another contextual computation F[A] that can use `F[Ctx]` inside.
+  *
+  * @tparam F context-aware effect e.g.`ReaderT[Lower, Ctx, *]`
+  */
 trait Provide[F[_]] extends ContextBase {
   type Ctx
+
+  /** Result effect type without any notion of context */
   type Lower[A]
 
+  /** Runs `fa` providing `ctx` to it.
+    *
+    * One can treat F as function of type `Ctx => Lower[A]` so this method applies it to `ctx`.
+    *
+    * @param fa Contextual computation
+    * @return Result of running fa and providing it context
+    */
   def runContext[A](fa: F[A])(ctx: Ctx): Lower[A]
+
+  /** Same as [[runContext]] but higher-kinded.
+    *
+    * @example {{{
+    *   trait ProcessHandler[G[_]] {
+    *     def mapK[M[_]](fk: G ~> M): ProcessHandler[M] = ???
+    *     //...other methods
+    *   }
+    *   type WithMyContext[F[_], A] = ReaderT[F, MyCtx, A]
+    *   val processHandler: ProcessHandler[IO WithMyContext *] = ???
```

Comment:
I think you should swap names. `ProcessHandler[IO WithContext *]` is contextual and `ProcessHandler[IO]` is not because you created it with constant context so you can't change it, providing after method calls.